yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,362 @@
+    }, {
+        'url': 'https://bsky.app/profile/alt.bun.how/post/3l7rdfxhyds2f',
+        'md5': '8775118b235cf9fa6b5ad30f95cda75c',
+        'info_dict': {
+            'id': '3l7rdfxhyds2f',
+            'ext': 'mp4',
+            'channel_url': 'https://bsky.app/profile/did:plc:7x6rtuenkuvxq3zsvffp2ide',
+            'timestamp': 1730332128,
+            'channel_id': 'did:plc:7x6rtuenkuvxq3zsvffp2ide',
+            'upload_date': '20241030',
+            'channel': 'alt.bun.how',
+            'uploader_id': 'did:plc:7x6rtuenkuvxq3zsvffp2ide',
+            'description': 'crazy that i look like this tbh',
+            'comment_count': int,
+            'thumbnail': r're:https://video.bsky.app/watch/.*\.jpg$',
+            'uploader_url': 'https://bsky.app/profile/alt.bun.how',
+            'tags': ['en', 'sexual', 'sexual'],
+            'like_count': int,
+            'title': 'cinnamon on Bluesky',
+            'uploader': 'cinnamon',
+            'repost_count': int,
+            'age_limit': 18,
+        },
+    }]
+
+    def _get_comments(self, meta):
+        yield from self.traverse_replies(meta, traverse_obj(meta, ('post', 'uri'), default=''))
+
+    def traverse_replies(self, thread_node, root_uri):
+        post_uri = traverse_obj(thread_node, ('post', 'uri'))
+        if post_uri != root_uri:
+            post = thread_node.get('post')
+            parent_uri = traverse_obj(post, ('record', 'reply', 'parent', 'uri'))
+            author_handle = traverse_obj(post, ('author', 'handle'))
+            author_did = traverse_obj(post, ('author', 'did'), default='')
+            yield {
+                'id': post_uri,
+                **traverse_obj(post, {
+                    'text': ('record', 'text'),
+                    'timestamp': ('record', 'createdAt', {parse_iso8601}),
+                    'author': ('author', 'displayName'),
+                    'author_thumbnail': ('author', 'avatar', {url_or_none}),
+                }),
+                'parent': 'root' if parent_uri == root_uri else parent_uri,
+                'like_count': post.get('likeCount'),
+                'author_id': author_did,
+                'author_url': f'https://bsky.app/profile/{author_handle}',
+                'author_is_uploader': author_did in root_uri,
+            }
+        if replies := thread_node.get('replies'):
+            for reply in replies:
+                yield from self.traverse_replies(reply, root_uri)
+        if parent := thread_node.get('parent'):
+            yield from self.traverse_replies(parent, root_uri)
+
+    def get_service_endpoint(self, did, video_id):
+        services = self._download_json(
+            f'https://resolver.identity.foundation/1.0/identifiers/{did}',
+            video_id, fatal=False).get('service') or []
+        for service in services:
+            if service.get('type') == 'AtprotoPersonalDataServer':
+                return service.get('serviceEndpoint')
+        return 'https://bsky.social'
+
```

Comment:
For the video extract method, this is what I came up with:
```suggestion

    @staticmethod
    def _build_profile_url(path):
        return format_field(path, None, 'https://bsky.app/profile/%s', default=None)

    def _extract_videos(self, root, video_id, embed_path='embed', record_path='record', record_subpath='embed'):
        embed_path = variadic(embed_path, (str, bytes, dict, set))
        record_path = variadic(record_path, (str, bytes, dict, set))
        record_subpath = variadic(record_subpath, (str, bytes, dict, set))

        entries = []
        if external_uri := traverse_obj(root, (((*record_path, *record_subpath), embed_path), 'external', 'uri', {url_or_none}, any)):
            entries.append(self.url_result(external_uri))

        if playlist := traverse_obj(root, (*embed_path, 'playlist', {url_or_none})):
            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
                playlist, video_id, 'mp4', m3u8_id='hls', fatal=False)
        else:
            return entries

        video_cid = traverse_obj(
            root, (*embed_path, 'cid', {str}),
            (*record_path, *record_subpath, 'video', 'ref', '$link', {str}))
        did = traverse_obj(root, ('author', 'did', {str}))
        if did and video_cid:
            endpoint = self.get_service_endpoint(did, video_id)
            formats.append({
                'format_id': 'blob',
                'url': update_url_query(self._BLOB_URL % endpoint, {'did': did, 'cid': video_cid}),
                **traverse_obj(root, (*embed_path, 'aspectRatio', {
                    'width': ('width', {int_or_none}),
                    'height': ('height', {int_or_none}),
                })),
                **traverse_obj(root, (*record_path, *record_subpath, 'video', {
                    'filesize': ('size', {int_or_none}),
                    'ext': ('mimeType', {mimetype2ext}),
                })),
            })
            for sub_data in traverse_obj(root, (
                    *record_path, *record_subpath, 'captions', lambda _, v: v['file']['ref']['$link'])):
                subtitles.setdefault(sub_data.get('lang') or 'und', []).append({
                    'url': update_url_query(self._BLOB_URL % endpoint, {'did': did, 'cid': sub_data['file']['ref']['$link']}),
                    'ext': traverse_obj(sub_data, ('file', 'mimeType', {mimetype2ext})),
                })

        entries.append({
            'id': video_id,
            'formats': formats,
            'subtitles': subtitles,
            **traverse_obj(root, {
                'id': ('uri', {lambda x: x.split('/')[-1] or None}, {str}),
                'thumbnail': (*embed_path, 'thumbnail', {url_or_none}),
                'alt_title': (*embed_path, 'alt', {str}),
                'uploader': ('author', 'displayName', {str}),
                'uploader_id': ('author', 'handle', {str}),
                'uploader_url': ('author', 'handle', {self._build_profile_url}),
                'channel_id': ('author', 'did', {str}),
                'channel_url': ('author', 'did', {self._build_profile_url}),
                'like_count': ('likeCount', {int_or_none}),
                'repost_count': ('repostCount', {int_or_none}),
                'comment_count': ('replyCount', {int_or_none}),
                'timestamp': ('indexedAt', {parse_iso8601}),
                'tags': ('labels', ..., 'val', {str}, all, {lambda x: list(set(x))}),
                'age_limit': ('labels', ..., 'val', {lambda x: 18 if x in ('sexual', 'porn', 'graphic-media') else None}, any),
                'description': (*record_path, 'text', {str}),
                'title': (*record_path, 'text', {lambda x: truncate_string(x, 50)}),
            }),
        })
        return entries

```
This requires the const `_BLOB_URL = '%s/xrpc/com.atproto.sync.getBlob'` to be added.

Some notes on the implementation/differences to yours:
- I tried to combine the different paths with paths to the record/embed being able to be passed
- I adjusted the uploader/channel(_id/_url) logic. I think like this it makes sense, and is somewhat in line with twitter.
- For the tags, I removed the `language`. Potentially, we could return this in the `language` field. I also deduplicated the tags.
- For the title, I think it makes more sense to go with the same value as the description. To not have the same issue as with the twitter extractor (potentially very long filenames that are not supported by the target filesystem) I would truncate the string.